In [10]:
import sys
sys.path.append("..")

from data.getdata import get_data
from features.build_features import build_features
from model.train import trainModel


☁️  Downloading fresh data from Azure SQL...
✅ Data saved to lance_loans_cache.parquet


In [11]:
df = get_data()
df = build_features(df)

df.shape  # sanity check


⚡ Loading from local cache (Fast)...


(119527, 70)

In [12]:
model, roc_auc, recall = trainModel(df)


ROC-AUC: 0.8071
Recall (defaulters): 0.6788


ROC-AUC = 0.8071

In ~81% of random defaulter vs non-defaulter pairs, the model assigns a higher rist score to the defaulter

Recall (Defaultes) = 0.6788

We are catching ~68% of random defaulters at the current threshold.

Given:
 Default rate : 2.2%
 Threshold = 0.5 (almost certainly)

Solud baseline though

In [13]:
#Inspecting coefficients
coef_df = pd.DataFrame({
    "feature": model.feature_names_in_,
    "coef": model.coef_[0]
}).sort_values("coef", ascending=False)

coef_df.head(10)   # top risk drivers
coef_df.tail(10)   # strongest protections


feature      coef
40                    res_type_RENT -0.154287
26      Product_code_two_wheeler_SC -0.168534
43                    Tier_'TIER 3' -0.197113
36               Employment_STUDENT -0.201245
21               num_60d_pastDue_6m -0.228921
61  sanctioned_amt_sercured_missing -0.299171
22               num_90d_pastDue_3m -0.325480
65               first_loan_missing -0.325672
27      Product_code_two_wheeler_TL -0.642947
64                last_loan_missing -0.683822